daisy/image.go

```diff
@@ -80,20 +73,24 @@ func imageExists(client daisyCompute.Client, project, family, name string) (bool
 	imageCache.mu.Lock()
 	defer imageCache.mu.Unlock()
 	if imageCache.exists == nil {
-		imageCache.exists = map[string][]*compute.Image{}
+		imageCache.exists = map[string][]interface{}{}
 	}
 	if _, ok := imageCache.exists[project]; !ok {
 		il, err := client.ListImages(project)
 		if err != nil {
 			return false, Errf("error listing images for project %q: %v", project, err)
 		}
-		imageCache.exists[project] = il
+		ila := make([]interface{}, len(il))
+		for _, i := range il {
+			ila = append(ila, i)
+		}
+		imageCache.exists[project] = ila
 	}
 
 	for _, i := range imageCache.exists[project] {
-		if name == i.Name {
-			if i.Deprecated != nil && (i.Deprecated.State == "OBSOLETE" || i.Deprecated.State == "DELETED") {
-				return true, typedErrf(imageObsoleteDeletedError, "image %q in state %q", name, i.Deprecated.State)
+		if ic, ok := i.(*compute.Image); ok && name == ic.Name {
```

Comment:
What is this cast for? Guard against alpha/beta objects?

Reply:
The type before this PR was *compute.Image. Since it was changed to interface{}, we have to cast bgefore using its attributes "ic.Name"